packages/react-reconciler/src/ReactStrictModeWarnings.js

```diff
@@ -129,175 +42,260 @@ if (__DEV__) {
+          'and may indicate bugs in your code. ' +
+          'See https://fb.me/react-async-component-lifecycle-hooks for details.\n\n' +
+          '* Move data fetching code or side effects to componentDidUpdate.\n' +
+          "* If you're updating state whenever props change, " +
+          'refactor your code to use memoization techniques, or move it to ' +
+          'static getDerivedStateFromProps. Learn more at: https://fb.me/react-derived-state\n' +
+          '\nPlease update the following components: %s',
+        sortedNames,
+      );
     }
 
-    let warningsForRoot;
-    if (!pendingUnsafeLifecycleWarnings.has(strictRoot)) {
-      warningsForRoot = {
-        UNSAFE_componentWillMount: [],
-        UNSAFE_componentWillReceiveProps: [],
-        UNSAFE_componentWillUpdate: [],
-      };
-
-      pendingUnsafeLifecycleWarnings.set(strictRoot, warningsForRoot);
-    } else {
-      warningsForRoot = pendingUnsafeLifecycleWarnings.get(strictRoot);
+    if (UNSAFE_componentWillUpdateUniqueNames.size > 0) {
+      const sortedNames = setToSortedString(
+        UNSAFE_componentWillUpdateUniqueNames,
+      );
+      warningWithoutStack(
+        false,
+        'Using UNSAFE_componentWillUpdate in strict mode is not recommended ' +
+          'and may indicate bugs in your code. ' +
+          'See https://fb.me/react-async-component-lifecycle-hooks for details.\n\n' +
+          '* Move data fetching code or side effects to componentDidUpdate.\n' +
+          '\nPlease update the following components: %s',
+        sortedNames,
+      );
     }
 
-    const unsafeLifecycles = [];
-    if (
-      (typeof instance.componentWillMount === 'function' &&
-        instance.componentWillMount.__suppressDeprecationWarning !== true) ||
-      typeof instance.UNSAFE_componentWillMount === 'function'
-    ) {
-      unsafeLifecycles.push('UNSAFE_componentWillMount');
-    }
-    if (
-      (typeof instance.componentWillReceiveProps === 'function' &&
-        instance.componentWillReceiveProps.__suppressDeprecationWarning !==
-          true) ||
-      typeof instance.UNSAFE_componentWillReceiveProps === 'function'
-    ) {
-      unsafeLifecycles.push('UNSAFE_componentWillReceiveProps');
+    if (componentWillMountUniqueNames.size > 0) {
+      const sortedNames = setToSortedString(componentWillMountUniqueNames);
+
+      lowPriorityWarning(
+        false,
+        'componentWillMount has been renamed, and is not recommended for use. ' +
+          'See https://fb.me/react-async-component-lifecycle-hooks for details.\n\n' +
+          '* Move code with side effects to componentDidMount, and set initial state in the constructor.\n' +
+          '* Rename componentWillMount to UNSAFE_componentWillMount to suppress ' +
+          'this warning in non-strict mode. In React 17.x, only the UNSAFE_ name will work.\n' +
+          '* To rename all deprecated lifecycles to their new names, you can run ' +
```

Comment:
I still think these two bullets should be combined. Separating them is confusing.

Reply:
I don't feel as strongly about this, so I'm happy to combine them. Will update.